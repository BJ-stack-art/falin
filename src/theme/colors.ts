export const neutral = {
  25: "#f2f4f7",
  50: "#f9fafb",
  100: "#f2f4f7",
  200: "#eaecf0",
  300: "#d0d5dd",
  400: "#98a2b3",
  500: "#667085",
  600: "#475467",
  700: "#344054",
  800: "#1D2939",
  900: "#101828",
};

export const primary = {
  25: "#f5f8ff",
  50: "#eff4ff",
  100: "#d1e0ff",
  200: "#b2ccff",
  300: "#84adff",
  400: "#528bff",
  500: "#2970ff",
  600: "#155eef",
  700: "#004eeb",
  800: "#0040c1",
  900: "#00359e",
};

export const success = {
  25: "#f6fef9",
  50: "#ecfdf3",
  100: "#d1fadf",
  200: "#a6f4c5",
  300: "#6ce9a6",
  400: "#32d583",
  500: "#12b76a",
  600: "#039855",
  700: "#027a48",
  800: "#05603a",
  900: "#054f31",
};

export const info = {
  25: "#fafeff",
  50: "#f0fcff",
  100: "#c6fffa",
  200: "#91fefe",
  300: "#5ff2fe",
  400: "#3cdffd",
  500: "#13c1fc",
  600: "#0d92d3",
  700: "#086bab",
  800: "#044b85",
  900: "#044b85",
};

export const warning = {
  25: "#fffcf5",
  50: "#fffaeb",
  100: "#fef0c7",
  200: "#fedf89",
  300: "#fec84b",
  400: "#fdb022",
  500: "#f79009",
  600: "#dc6803",
  700: "#b54708",
  800: "#93370d",
  900: "#7a2e0e",
};

export const danger = {
  25: "#fffbfa",
  50: "#fef3f2",
  100: "#fee4e2",
  200: "#fecdca",
  300: "#fda29b",
  400: "#f97066",
  500: "#f04438",
  600: "#d92d20",
  700: "#b42318",
  800: "#912018",
  900: "#7a271a",
};
