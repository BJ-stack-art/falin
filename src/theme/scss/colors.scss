$neutral-25: #f2f4f7;
$neutral-50: #f8fafc;
$neutral-100: #f1f5f9;
$neutral-200: #e2e8f0;
$neutral-300: #cbd5e1;
$neutral-400: #94a3b8;
$neutral-500: #64748b;
$neutral-600: #475569;
$neutral-700: #334155;
$neutral-800: #1e293b;
$neutral-900: #0f172a;

$primary-25: #f5f8ff;
$primary-50: #eff6ff;
$primary-100: #dbeafe;
$primary-200: #bfdbfe;
$primary-300: #93c5fd;
$primary-400: #60a5fa;
$primary-500: #3b82f6;
$primary-600: #2563eb;
$primary-700: #1d4ed8;
$primary-800: #1e40af;
$primary-900: #1e3a8a;

$success-25: #f6fef9;
$success-50: #ecfdf3;
$success-100: #d1fadf;
$success-200: #a6f4c5;
$success-300: #6ce9a6;
$success-400: #32d583;
$success-500: #12b76a;
$success-600: #039855;
$success-700: #027a48;
$success-800: #05603a;
$success-900: #054f31;

$info-25: #fafeff;
$info-50: #f0f9ff;
$info-100: #e0f2fe;
$info-200: #bae6fd;
$info-300: #7dd3fc;
$info-400: #38bdf8;
$info-500: #0ea5e9;
$info-600: #0284c7;
$info-700: #0369a1;
$info-800: #075985;
$info-900: #0c4a6e;

$warning-25: #fffcf5;
$warning-50: #fff7ed;
$warning-100: #ffedd5;
$warning-200: #fed7aa;
$warning-300: #fdba74;
$warning-400: #fb923c;
$warning-500: #f97316;
$warning-600: #ea580c;
$warning-700: #c2410c;
$warning-800: #9a3412;
$warning-900: #7c2d12;

$danger-25: #fffbfa;
$danger-50: #fef2f2;
$danger-100: #fee2e2;
$danger-200: #fecaca;
$danger-300: #fca5a5;
$danger-400: #f87171;
$danger-500: #ef4444;
$danger-600: #dc2626;
$danger-700: #b91c1c;
$danger-800: #991b1b;
$danger-900: #7f1d1d;
