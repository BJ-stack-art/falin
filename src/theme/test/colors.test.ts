import { describe, expect, test } from "vitest"
import { danger, info, neutral, primary, success, warning } from "../colors"
import { readFileSync } from "fs"

describe("Theme Color", () => {
  test("should match neutral snapshot", () => {
    expect(neutral).toMatchSnapshot()
  })

  test("should match primary snapshot", () => {
    expect(primary).toMatchSnapshot()
  })

  test("should match success snapshot", () => {
    expect(success).toMatchSnapshot()
  })

  test("should match info snapshot", () => {
    expect(info).toMatchSnapshot()
  })

  test("should match warning snapshot", () => {
    expect(warning).toMatchSnapshot()
  })

  test("should match danger snapshot", () => {
    expect(danger).toMatchSnapshot()
  })

  test("should match scss variables snapshot", () => {
    expect(readFileSync(__dirname + "./../scss/colors.scss")).toMatchSnapshot()
  })
})
