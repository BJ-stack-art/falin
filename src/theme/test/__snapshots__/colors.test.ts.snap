// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Theme Color > should match danger snapshot 1`] = `
{
  "100": "#fee4e2",
  "200": "#fecdca",
  "25": "#fffbfa",
  "300": "#fda29b",
  "400": "#f97066",
  "50": "#fef3f2",
  "500": "#f04438",
  "600": "#d92d20",
  "700": "#b42318",
  "800": "#912018",
  "900": "#7a271a",
}
`;

exports[`Theme Color > should match info snapshot 1`] = `
{
  "100": "#c6fffa",
  "200": "#91fefe",
  "25": "#fafeff",
  "300": "#5ff2fe",
  "400": "#3cdffd",
  "50": "#f0fcff",
  "500": "#13c1fc",
  "600": "#0d92d3",
  "700": "#086bab",
  "800": "#044b85",
  "900": "#044b85",
}
`;

exports[`Theme Color > should match neutral snapshot 1`] = `
{
  "100": "#f2f4f7",
  "200": "#eaecf0",
  "25": "#f9fafb",
  "300": "#d0d5dd",
  "400": "#98a2b3",
  "50": "#f2f4f7",
  "500": "#667085",
  "600": "#475467",
  "700": "#344054",
  "800": "#1D2939",
  "900": "#101828",
}
`;

exports[`Theme Color > should match primary snapshot 1`] = `
{
  "100": "#d1e0ff",
  "200": "#b2ccff",
  "25": "#f5f8ff",
  "300": "#84adff",
  "400": "#528bff",
  "50": "#eff4ff",
  "500": "#2970dd",
  "600": "#155eef",
  "700": "#004eeb",
  "800": "#0040c1",
  "900": "#00359e",
}
`;

exports[`Theme Color > should match scss variables snapshot 1`] = `
{
  "data": [
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    50,
    53,
    58,
    32,
    35,
    102,
    50,
    102,
    52,
    102,
    55,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    53,
    48,
    58,
    32,
    35,
    102,
    57,
    102,
    97,
    102,
    98,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    49,
    48,
    48,
    58,
    32,
    35,
    102,
    50,
    102,
    52,
    102,
    55,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    50,
    48,
    48,
    58,
    32,
    35,
    101,
    97,
    101,
    99,
    102,
    48,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    51,
    48,
    48,
    58,
    32,
    35,
    100,
    48,
    100,
    53,
    100,
    100,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    52,
    48,
    48,
    58,
    32,
    35,
    57,
    56,
    97,
    50,
    98,
    51,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    53,
    48,
    48,
    58,
    32,
    35,
    54,
    54,
    55,
    48,
    56,
    53,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    54,
    48,
    48,
    58,
    32,
    35,
    52,
    55,
    53,
    52,
    54,
    55,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    55,
    48,
    48,
    58,
    32,
    35,
    51,
    52,
    52,
    48,
    53,
    52,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    56,
    48,
    48,
    58,
    32,
    35,
    49,
    100,
    50,
    57,
    51,
    57,
    59,
    13,
    10,
    36,
    110,
    101,
    117,
    116,
    114,
    97,
    108,
    45,
    57,
    48,
    48,
    58,
    32,
    35,
    49,
    48,
    49,
    56,
    50,
    56,
    59,
    13,
    10,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    50,
    53,
    58,
    32,
    35,
    102,
    53,
    102,
    56,
    102,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    53,
    48,
    58,
    32,
    35,
    101,
    102,
    102,
    52,
    102,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    49,
    48,
    48,
    58,
    32,
    35,
    100,
    49,
    101,
    48,
    102,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    50,
    48,
    48,
    58,
    32,
    35,
    98,
    50,
    99,
    99,
    102,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    51,
    48,
    48,
    58,
    32,
    35,
    56,
    52,
    97,
    100,
    102,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    52,
    48,
    48,
    58,
    32,
    35,
    53,
    50,
    56,
    98,
    102,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    53,
    48,
    48,
    58,
    32,
    35,
    50,
    57,
    55,
    48,
    102,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    54,
    48,
    48,
    58,
    32,
    35,
    49,
    53,
    53,
    101,
    101,
    102,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    55,
    48,
    48,
    58,
    32,
    35,
    48,
    48,
    52,
    101,
    101,
    98,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    56,
    48,
    48,
    58,
    32,
    35,
    48,
    48,
    52,
    48,
    99,
    49,
    59,
    13,
    10,
    36,
    112,
    114,
    105,
    109,
    97,
    114,
    121,
    45,
    57,
    48,
    48,
    58,
    32,
    35,
    48,
    48,
    51,
    53,
    57,
    101,
    59,
    13,
    10,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    50,
    53,
    58,
    32,
    35,
    102,
    54,
    102,
    101,
    102,
    57,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    53,
    48,
    58,
    32,
    35,
    101,
    99,
    102,
    100,
    102,
    51,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    49,
    48,
    48,
    58,
    32,
    35,
    100,
    49,
    102,
    97,
    100,
    102,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    50,
    48,
    48,
    58,
    32,
    35,
    97,
    54,
    102,
    52,
    99,
    53,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    51,
    48,
    48,
    58,
    32,
    35,
    54,
    99,
    101,
    57,
    97,
    54,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    52,
    48,
    48,
    58,
    32,
    35,
    51,
    50,
    100,
    53,
    56,
    51,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    53,
    48,
    48,
    58,
    32,
    35,
    49,
    50,
    98,
    55,
    54,
    97,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    54,
    48,
    48,
    58,
    32,
    35,
    48,
    51,
    57,
    56,
    53,
    53,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    55,
    48,
    48,
    58,
    32,
    35,
    48,
    50,
    55,
    97,
    52,
    56,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    56,
    48,
    48,
    58,
    32,
    35,
    48,
    53,
    54,
    48,
    51,
    97,
    59,
    13,
    10,
    36,
    115,
    117,
    99,
    99,
    101,
    115,
    115,
    45,
    57,
    48,
    48,
    58,
    32,
    35,
    48,
    53,
    52,
    102,
    51,
    49,
    59,
    13,
    10,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    50,
    53,
    58,
    32,
    35,
    102,
    97,
    102,
    101,
    102,
    102,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    53,
    48,
    58,
    32,
    35,
    102,
    48,
    102,
    99,
    102,
    102,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    49,
    48,
    48,
    58,
    32,
    35,
    99,
    54,
    102,
    102,
    102,
    97,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    50,
    48,
    48,
    58,
    32,
    35,
    57,
    49,
    102,
    101,
    102,
    101,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    51,
    48,
    48,
    58,
    32,
    35,
    53,
    102,
    102,
    50,
    102,
    101,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    52,
    48,
    48,
    58,
    32,
    35,
    51,
    99,
    100,
    102,
    102,
    100,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    53,
    48,
    48,
    58,
    32,
    35,
    49,
    51,
    99,
    49,
    102,
    99,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    54,
    48,
    48,
    58,
    32,
    35,
    48,
    100,
    57,
    50,
    100,
    51,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    55,
    48,
    48,
    58,
    32,
    35,
    48,
    56,
    54,
    98,
    97,
    98,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    56,
    48,
    48,
    58,
    32,
    35,
    48,
    52,
    52,
    98,
    56,
    53,
    59,
    13,
    10,
    36,
    105,
    110,
    102,
    111,
    45,
    57,
    48,
    48,
    58,
    32,
    35,
    48,
    52,
    52,
    98,
    56,
    53,
    59,
    13,
    10,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    50,
    53,
    58,
    32,
    35,
    102,
    102,
    102,
    99,
    102,
    53,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    53,
    48,
    58,
    32,
    35,
    102,
    102,
    102,
    97,
    101,
    98,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    49,
    48,
    48,
    58,
    32,
    35,
    102,
    101,
    102,
    48,
    99,
    55,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    50,
    48,
    48,
    58,
    32,
    35,
    102,
    101,
    100,
    102,
    56,
    57,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    51,
    48,
    48,
    58,
    32,
    35,
    102,
    101,
    99,
    56,
    52,
    98,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    52,
    48,
    48,
    58,
    32,
    35,
    102,
    100,
    98,
    48,
    50,
    50,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    53,
    48,
    48,
    58,
    32,
    35,
    102,
    55,
    57,
    48,
    48,
    57,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    54,
    48,
    48,
    58,
    32,
    35,
    100,
    99,
    54,
    56,
    48,
    51,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    55,
    48,
    48,
    58,
    32,
    35,
    98,
    53,
    52,
    55,
    48,
    56,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    56,
    48,
    48,
    58,
    32,
    35,
    57,
    51,
    51,
    55,
    48,
    100,
    59,
    13,
    10,
    36,
    119,
    97,
    114,
    110,
    105,
    110,
    103,
    45,
    57,
    48,
    48,
    58,
    32,
    35,
    55,
    97,
    50,
    101,
    48,
    101,
    59,
    13,
    10,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    50,
    53,
    58,
    32,
    35,
    102,
    102,
    102,
    98,
    102,
    97,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    53,
    48,
    58,
    32,
    35,
    102,
    101,
    102,
    51,
    102,
    50,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    49,
    48,
    48,
    58,
    32,
    35,
    102,
    101,
    101,
    52,
    101,
    50,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    50,
    48,
    48,
    58,
    32,
    35,
    102,
    101,
    99,
    100,
    99,
    97,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    51,
    48,
    48,
    58,
    32,
    35,
    102,
    100,
    97,
    50,
    57,
    98,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    52,
    48,
    48,
    58,
    32,
    35,
    102,
    57,
    55,
    48,
    54,
    54,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    53,
    48,
    48,
    58,
    32,
    35,
    102,
    48,
    52,
    52,
    51,
    56,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    54,
    48,
    48,
    58,
    32,
    35,
    100,
    57,
    50,
    100,
    50,
    48,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    55,
    48,
    48,
    58,
    32,
    35,
    98,
    52,
    50,
    51,
    49,
    56,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    56,
    48,
    48,
    58,
    32,
    35,
    57,
    49,
    50,
    48,
    49,
    56,
    59,
    13,
    10,
    36,
    100,
    97,
    110,
    103,
    101,
    114,
    45,
    57,
    48,
    48,
    58,
    32,
    35,
    55,
    97,
    50,
    55,
    49,
    97,
    59,
    13,
    10,
  ],
  "type": "Buffer",
}
`;

exports[`Theme Color > should match success snapshot 1`] = `
{
  "100": "#d1fadf",
  "200": "#a6f4c5",
  "25": "#f6fef9",
  "300": "#6ce9a6",
  "400": "#32d583",
  "50": "#ecfdf3",
  "500": "#12b76a",
  "600": "#039855",
  "700": "#027a48",
  "800": "#05603a",
  "900": "#054f31",
}
`;

exports[`Theme Color > should match warning snapshot 1`] = `
{
  "100": "#fef0c7",
  "200": "#fedf89",
  "25": "#fffcf5",
  "300": "#fec84b",
  "400": "#fdb022",
  "50": "#fffaeb",
  "500": "#f79009",
  "600": "#dc6803",
  "700": "#b54708",
  "800": "#93370d",
  "900": "#7a2e0e",
}
`;
